libs/core/langchain_core/vectorstores/base.py

```diff
@@ -827,6 +836,15 @@ def from_documents(
         """
         texts = [d.page_content for d in documents]
         metadatas = [d.metadata for d in documents]
+
+        if "ids" not in kwargs:
```

Comment:
Should we add deprecation warnings that `ids` should be passed in `Document` and not as a param ?

Reply:
The optional id in the document is optional now and not all implementations honor them. (if I had to guess most don't yet)

So far I've been interpreting these as precedence rules -- ids specified as part of the function signature take precedence over ids specified as part of the document.